pkg/pkg.mk

```diff
@@ -79,6 +79,11 @@ PKG_DOWNLOADED = $(PKG_STATE)-downloaded
 # Custom prepared target that can be defined in packages Makefile.
 PKG_CUSTOM_PREPARED ?=
 
+# Custom post-patched target that can be defined in packages Makefile.
+# this is useful if something more complex than a static patch can address
+# needs to be done *after checkout*
+PKG_CUSTOM_POST_PATCH ?=
```

Comment:
Do you have an example for that?

Reply:
@benpicco  I just added a trivial example to the u8g2 package - in the case of when the `u8g2_sdl` package is required, it tests for the command `sdl2-config --version` to make sure it exists.  But if you run it again, it won't run that test, because the `touch` command put a .pkg file in the build folder, like the downloaded, patched operations already do.

Tested in tests/pkg/test_u8g2:

First run:
```
make BOARD=native TEST_OUTPUT=2
Building application "tests_u8g2" for "native" with MCU "native".

Testing for SDL utilities
"make" -C /home/devuser/workspace/bissell_riot/pkg/u8g2/ 
make[1]: '/home/devuser/workspace/bissell_riot/build/pkg/u8g2/.pkg-state.git-custom-post-patched' is up to date.
```

If you do another run without cleaning build directory out: 
```
make BOARD=native TEST_OUTPUT=2
Building application "tests_u8g2" for "native" with MCU "native".

"make" -C /home/devuser/workspace/bissell_riot/pkg/u8g2/ 
make[1]: '/home/devuser/workspace/bissell_riot/build/pkg/u8g2/.pkg-state.git-custom-post-patched' is up to date.
```

You'll notice the first one got the "Testing for SDL Utilities" print and the subsequent call did not.  This is of course up to the author of the package makefile - if they exclude that `touch $@` line, it would run that build target on every call to `make`.

Reply:
The example is a bit weird as it only checks once - I wouldn't necessarily commit that.

I was more thinking of an example to mention in the comment, so people can have an idea what this might be used for.

Reply:
@benpicco how does that look?